libs/api/domains/vehicles/src/lib/api-domains-vehicles-mileage.resolver.ts

```diff
@@ -0,0 +1,100 @@
+import {
+  Args,
+  Context,
+  Mutation,
+  Parent,
+  Query,
+  ResolveField,
+  Resolver,
+} from '@nestjs/graphql'
+import { UseGuards } from '@nestjs/common'
+import {
+  IdsUserGuard,
+  ScopesGuard,
+  Scopes,
+  CurrentUser,
+} from '@island.is/auth-nest-tools'
+import type { User } from '@island.is/auth-nest-tools'
+import { ApiScope } from '@island.is/auth/scopes'
+import { Audit } from '@island.is/nest/audit'
+import { VehiclesService } from './api-domains-vehicles.service'
+import {
+  VehicleMileageDetail,
+  VehicleMileageOverview,
+  VehicleMileagePutModel,
+} from '../models/getVehicleMileage.model'
+import { GetVehicleMileageInput } from '../dto/getVehicleMileageInput'
+import {
+  PostVehicleMileageInput,
+  PutVehicleMileageInput,
+} from '../dto/postVehicleMileageInput'
+
+@UseGuards(IdsUserGuard, ScopesGuard)
+@Resolver(() => VehicleMileageOverview)
+@Audit({ namespace: '@island.is/api/vehicles' })
+export class VehiclesMileageResolver {
+  constructor(private readonly vehiclesService: VehiclesService) {}
+
+  @Scopes(ApiScope.vehicles)
```

Comment:
~ The `@Scopes` decorator can be on the class level so all methods are protected by it by default.